### PR TITLE
Selector: further default to Linux behaviour

### PR DESCRIPTION
**Pull request by compnerd**
This furthers the behaviour uniformity of special casing behaviour for
other platforms (e.g. Darwin, Windows) and defaulting to the Linux
behaviour as the rest of the code base assumes that the Linux behaviour
is a reasonable fallback.  Ensuring that the paths taken are always
uniformly taken helps identify the changes necessary for a port and make
it easier to reason about the changes.

_[One line description of your change]_

### Motivation:

_[Explain here the context, and why you're making that change. What is the problem you're trying to solve.]_

### Modifications:

_[Describe the modifications you've done.]_

### Result:

_[After your change, what will change.]_

### CODE REVIEW

**Comment by compnerd** [comment]
Sure, that’s not a problem to change.  However there’s a mixture in the code, should everything be reindented?

**Comment by Lukasa** [comment]
Let's only do this for the ones we're changing: no need to add noise to this diff.

**Comment by compnerd** [comment]
Oh, missed the bit about the additional indentation level.

**Comment by compnerd** [comment]
Okay, reformatted to put the `#if` at the code level, code itself indented an additional level for the places where I've changed things, the rest of the file still has a few different styles.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by compnerd** [comment]
Anything else left for this one?

File: Sources/NIO/Selector.swift [hunk 1]
```diff
@@ -663,26 +662,26 @@ internal class Selector<R: Registration> {
...
-            event.filter = Int16(EVFILT_USER)
-            event.fflags = UInt32(NOTE_TRIGGER | NOTE_FFNOP)
-            event.data = 0
-            event.udata = nil
-            event.flags = 0
-            try withUnsafeMutablePointer(to: &event) { ptr in
-                try self.kqueueApplyEventChangeSet(keventBuffer: UnsafeMutableBufferPointer(start: ptr, count: 1))
-            }
-        #endif
+            #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
```
Comment (weissi) [inline Sources/NIO/Selector.swift]: nit: indentation isn't quite right here.

**Pull request merged**